Privacy Dashboard 8.1.0: Improved breakage form

mgurgel opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before working on a PR, as we can't guarantee that we will accept your changes if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL: https://app.asana.com/0/0/1209127604215384/f
CC: @jaceklyp 

**Description**:

Upgrades Privacy Dashboard to 8.0.0, which contains the new site breakage flow as can be previewed [here](https://duckduckgo.github.io/privacy-dashboard/app-debug/html/iframe.html?platforms=android%2Cios%2Cmacos%2Cwindows%2Cbrowser&screen=breakageForm)

Figma: https://www.figma.com/design/eeVdakW2pjijbDlv8HUSgZ/O-E---Report-Broken-Site-Flow-(Design-Updates-2024-09)?node-id=4024-6272&m=dev

**Steps to test this PR**:

1. Trigger the breakage form in the two possible ways:
- Open the Privacy Dashboard and click on "Report a problem with this site"
- Open the app menu ••• and click on "Report Broken Site” (iOS) 

2. Confirm that the breakage form matches the Figma or the [web preview ](https://duckduckgo.github.io/privacy-dashboard/app-debug/html/iframe.html?platforms=android%2Cios%2Cmacos%2Cwindows%2Cbrowser&screen=breakageForm)

3. Attempt to complete a broken site report from both entry points above. Things to keep an eye on:
- The “See what’s sent” toggle expands and collapses normally
- When the category is “Something else”, trying to submit the form without a description should raise an alert
- The description is optional for all other categories


<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

jaceklyp committed:
Remove experiment code

mgurgel committed:
Updated packages

jaceklyp committed:
Remove experimental code

jaceklyp committed:
Remove experimental code

jaceklyp committed:
Privacy Dashboard 8.0.0

afterxleep reviewed (approved):
Working as described.

![01-15 at 16 32](https://github.com/user-attachments/assets/8e9b073b-5866-43dd-ac0b-16b96598701a)

mgurgel committed:
Aligning Privacy Dashboard EventMapping with macOS (#3822)

<!--
Note: This checklist is a reminder of our shared engineering
expectations. Feel free to change it, although assigning a GitHub
reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before
working on a PR, as we can't guarantee that we will accept your changes
if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL:
https://app.asana.com/0/1206594217596623/1209173355503842/f

**Description**:

Removes automatic firing of `BrokenSiteReportShown` pixel to align with
macOS. The firing of that pixel is now actively requested by the Privacy
Dashboard (see
https://github.com/duckduckgo/BrowserServicesKit/pull/1167)

This PR contains a fix for on
https://github.com/duckduckgo/iOS/pull/3802 which is why it is stacked
on it. Please let me know if this is not the best way to have this code
reviewed.

**Steps to test this PR**:

1. Invoke the breakage form in the two possible ways:
Open the Privacy Dashboard and click on "Report a problem with this
site"
Open the app menu ••• and click on "Report Broken Site” (iOS)

2. Confirm that the pixel `m_report-broken-site_shown` was fired
3. Submit a report
4. Confirm that the pixels `epbf` and `m_report-broken-site_sent` were
fired

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of
Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `’`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

—
[Software Engineering
Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design
Template](https://app.asana.com/0/59792373528535/184709971311943)

---------

Co-authored-by: Jacek Łyp <jlyp@duckduckgo.com>

mgurgel committed:
BSK bump
